internal/certdb/validation.go

```diff
@@ -27,27 +30,58 @@ func ValidateCertificateRequest(csr string) error {
 }
 
 // ValidateCertificate validates the given Cert string to the following:
-// The cert string must be a valid PEM string, and should be of type CERTIFICATE
-// The PEM string should be able to be parsed into a x509 Certificate
+//
+// The string must include 2 or more PEM formatted certificate strings.
+// Each cert must be a valid PEM string, and should be capable of being parsed into type x509 CERTIFICATE
+// Each subsequent certificate in the string should be the issuer of the previous string, which means:
+//
+// All except the first certificate should have the "is a CA" Basic Constraint.
+// The public key of the certificate should match the public key of the following certificate.
+// The issuer field of the certificate should match the subject field of the following certificate.
 func ValidateCertificate(cert string) error {
-	certBlock, _ := pem.Decode([]byte(cert))
-	if certBlock == nil {
-		return errors.New("PEM Certificate string not found or malformed")
+	certData := []byte(cert)
+	certificates := []*x509.Certificate{}
+	for {
+		certBlock, rest := pem.Decode(certData)
+		if certBlock == nil {
+			break
+		}
+		if certBlock.Type != "CERTIFICATE" {
+			return errors.New("a given PEM string was not a certificate")
+		}
+		certificate, err := x509.ParseCertificate(certBlock.Bytes)
+		if err != nil {
+			return err
+		}
+		certificates = append(certificates, certificate)
+		certData = rest
 	}
-	if certBlock.Type != "CERTIFICATE" {
-		return errors.New("given PEM string not a certificate")
+	if len(certificates) < 2 {
+		return errors.New("less than 2 certificate PEM strings were found")
 	}
-	_, err := x509.ParseCertificate(certBlock.Bytes)
-	if err != nil {
-		return err
+	for i := 0; i < len(certificates)-1; i++ {
+		firstCert := certificates[i]
+		secondCert := certificates[i+1]
+		if !bytes.Equal(firstCert.RawIssuer, secondCert.RawSubject) {
+			return fmt.Errorf("invalid certificate chain: certificate %d, certificate %d: subjects do not match", i, i+1)
+		}
+		if err := firstCert.CheckSignatureFrom(secondCert); err != nil {
+			return fmt.Errorf("invalid certificate chain: certificate %d, certificate %d: keys do not match: %s", i, i+1, err.Error())
+		}
+	}
+	for i := 1; i < len(certificates)-1; i++ {
+		cert := certificates[i]
```

Comment:
```suggestion
	for i, cert := range certificates[:len(certificates)-1] {
```

Reply:
this one actually starts at 1, so i used a range loop with the slice starting from 1.